lucene/packaging/build.gradle

```diff
@@ -0,0 +1,171 @@
+}
+
+description = 'Lucene distribution packaging'
+
+// Declare all subprojects that should be included in binary distribution.
+// By default everything is included, unless explicitly excluded.
+def includeInBinaries = project(":lucene").subprojects.findAll {subproject ->
+    return !(subproject.path in [
+        // Exclude packaging, not relevant to binary distribution.
+        ":lucene:packaging",
+        // Exclude parent container project of analysis modules (no artifacts).
+        ":lucene:analysis"
+    ])
+}
+
+// Create a configuration to each subproject and add dependency.
+def binaryArtifactsConf = { Project prj ->
+    "dep-binary" + prj.path.replace(':', '-')
+}
+
+def allDepsConf = { Project prj ->
+    "dep-full" + prj.path.replace(':', '-')
+}
+
+configurations {
+    docs
+}
+
+for (Project includedProject : includeInBinaries) {
+    def confBinaries = binaryArtifactsConf(includedProject)
+    def confFull = allDepsConf(includedProject)
+    configurations.create(confBinaries)
+    configurations.create(confFull)
+    dependencies { DependencyHandler handler ->
+        // Just project binaries.
+        handler.add(confBinaries, project(path: includedProject.path, configuration: "packaging"))
+        // All project dependencies, including transitive dependencies from the runtime configuration.
+        handler.add(confFull, project(path: includedProject.path, configuration: "runtimeElements"), {
+            exclude group: "org.apache.lucene"
+
+            // Exclude these from all projects.
+            exclude group: "commons-logging"
+            exclude group: "org.slf4j"
+        })
+    }
+}
+
+dependencies {
+    docs project(path: ':lucene', configuration: 'docs')
+}
+
+distributions {
+    // The "main" distribution is the binary distribution.
+    // We should also add 'source' distribution at some point
+    // (we can't do it now as the build itself is tangled with Solr).
+    main {
+        distributionBaseName = 'lucene'
+
+        contents {
+            // Manually correct posix permissions (matters when packaging on Windows).
+            filesMatching(["**/*.sh", "**/*.bat"]) { copy ->
+                copy.setMode(0755)
+            }
+
+            // Root distribution files; these are cherry-picked manually.
+            from(project(':lucene').projectDir, {
+                include "CHANGES.txt"
+                include "JRE_VERSION_MIGRATION.md"
+                include "LICENSE.txt"
+                include "licenses/*"
+                include "MIGRATE.md"
+                include "NOTICE.txt"
+                include "README.md"
+                include "SYSTEM_REQUIREMENTS.md"
+            })
+
+            // A couple more missing README files
+            from(project(':lucene:analysis').projectDir) {
+                include "README.txt"
+                into 'analysis'
+            }
+
+            // Copy files from documentation output to 'docs'
+            from(configurations.docs, {
```

Comment:
where is this coming from?

Reply:
This is part of the PR, Uwe - here.
https://github.com/apache/lucene-solr/pull/1905/files#diff-d55d7b9fe9d94d4e2e5554e3c1108943R60-R63

I suggested using configuration-based copying instead of pointing at an explicit folder because it has several advantages. First, an artifact exported via configuration is configurable in the place where it logically belongs to. So you can put together a set of documentation artifacts in documentation configuration snippet and just depend on it when you're assembling something that uses those artifacts.

The second benefit is that dependencies on tasks assembling those artifacts are automatically resolved - you don't need to know anything beyond the name of the project and configuration from which you're coping. The rest is done by gradle. 

I think Mike's patch is fine here.

Reply:
I fixed it to shipthe correct documentation. There were also some usless dependencies that I removed.